src/main/java/org/springframework/data/mapping/model/AbstractPersistentProperty.java

```diff
@@ -103,11 +105,27 @@ public AbstractPersistentProperty(Property property, PersistentEntity<?, P> owne
 		this.field = property.getField().orElse(null);
 		this.wither = property.getWither().orElse(null);
 
-		if (setter == null && (field == null || Modifier.isFinal(field.getModifiers()))) {
-			this.immutable = true;
-		} else {
-			this.immutable = false;
-		}
+		this.immutable = setter == null && (field == null || Modifier.isFinal(field.getModifiers()));
+		this.readable = Lazy.of(() -> {
+
+			if (setter != null) {
```

Comment:
How does the presence of a setter make a property readable? 🤔

Reply:
If there's no setter, then there's no way to set the property. While reading from the database, we need to put the value somewhere.

Reply:
I am sorry, I totally misunderstood what `….isReadable()` is even supposed to mean and totally got it backwards. I guess it's been a while. 😅